cpp/include/raft/sparse/neighbors/detail/connect_components.cuh

```diff
@@ -96,6 +105,22 @@ struct FixConnectivitiesRedOp {
   DI value_t get_value(KVP& out) const { return out.value; }
 
   DI value_t get_value(value_t& out) const { return out; }
+
+  void gather(const raft::resources& handle, value_idx* map)
```

Comment:
Can we add a comment to the effect of: 

"Gather and scatter are necessary because this functor is used in connect_components, which rearranges the data internally. The gather and scatter ensure that `operator()` is still consistent after rearranging. "